NeosModLoader/Utility/EnumerableInjector.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NeosModLoader.Utility
+{
+    /// <summary>
+    /// Provides the ability to inject actions into the execution of an enumeration while transforming it.
+    /// </summary>
+    /// <typeparam name="TIn">The type of the original enumeration.</typeparam>
+    /// <typeparam name="TOut">The type of the transformed enumeration.</typeparam>
+    public class EnumerableInjector<TIn, TOut> : IEnumerable<TOut>
```

Comment:
Probably would be good to include an example of usage if we're going to have utility classes without any context?

Reply:
what would you think makes a good example? 
I'Ve got two usages of it so far, one with Postfix and one with PostItem

Reply:
Dunno, probably just a hella world style logging one that demonstrates what you can do with it?